x/ecocredit/server/marketplace/msg_buy_direct_test.go

```diff
@@ -0,0 +1,609 @@
+	require.NoError(s.t, err)
+
+	require.Equal(s.t, order.Id, id)
+}
+
+func (s *buyDirectSuite) ExpectSellOrderWithQuantity(a string) {
+	order, err := s.marketStore.SellOrderTable().Get(s.ctx, s.sellOrderId)
+	require.NoError(s.t, err)
+
+	require.Equal(s.t, order.Quantity, a)
+}
+
+func (s *buyDirectSuite) ExpectNoSellOrderWithId(a string) {
+	id, err := strconv.ParseUint(a, 10, 32)
+	require.NoError(s.t, err)
+
+	_, err = s.marketStore.SellOrderTable().Get(s.ctx, id)
+	require.EqualError(s.t, err, ormerrors.NotFound.Error())
+}
+
+func (s *buyDirectSuite) ExpectAliceBankBalance(a string) {
+	bankBalance, err := sdk.ParseCoinNormalized(a)
+	require.NoError(s.t, err)
+
+	require.Equal(s.t, bankBalance.Denom, s.aliceBankBalance.Denom)
+	require.Equal(s.t, bankBalance.Amount.String(), s.aliceBankBalance.Amount.String())
+}
+
+func (s *buyDirectSuite) ExpectBobBankBalance(a string) {
+	bankBalance, err := sdk.ParseCoinNormalized(a)
+	require.NoError(s.t, err)
+
+	require.Equal(s.t, bankBalance.Denom, s.bobBankBalance.Denom)
+	require.Equal(s.t, bankBalance.Amount.String(), s.bobBankBalance.Amount.String())
+}
+
+func (s *buyDirectSuite) ExpectAliceBatchBalance(a gocuke.DocString) {
+	expected := &coreapi.BatchBalance{}
+	err := jsonpb.UnmarshalString(a.Content, expected)
+	require.NoError(s.t, err)
+
+	batch, err := s.coreStore.BatchTable().GetByDenom(s.ctx, s.batchDenom)
+	require.NoError(s.t, err)
+
+	balance, err := s.coreStore.BatchBalanceTable().Get(s.ctx, s.alice, batch.Key)
+	require.NoError(s.t, err)
+
+	require.Equal(s.t, expected.RetiredAmount, balance.RetiredAmount)
+	require.Equal(s.t, expected.TradableAmount, balance.TradableAmount)
+	require.Equal(s.t, expected.EscrowedAmount, balance.EscrowedAmount)
+}
+
+func (s *buyDirectSuite) ExpectBobBatchBalance(a gocuke.DocString) {
+	expected := &coreapi.BatchBalance{}
+	err := jsonpb.UnmarshalString(a.Content, expected)
+	require.NoError(s.t, err)
+
+	batch, err := s.coreStore.BatchTable().GetByDenom(s.ctx, s.batchDenom)
+	require.NoError(s.t, err)
+
+	balance, err := s.coreStore.BatchBalanceTable().Get(s.ctx, s.bob, batch.Key)
+	require.NoError(s.t, err)
+
+	require.Equal(s.t, expected.RetiredAmount, balance.RetiredAmount)
+	require.Equal(s.t, expected.TradableAmount, balance.TradableAmount)
+	require.Equal(s.t, expected.EscrowedAmount, balance.EscrowedAmount)
+}
+
+func (s *buyDirectSuite) ExpectBatchSupply(a gocuke.DocString) {
+	expected := &coreapi.BatchSupply{}
+	err := jsonpb.UnmarshalString(a.Content, expected)
+	require.NoError(s.t, err)
+
+	batch, err := s.coreStore.BatchTable().GetByDenom(s.ctx, s.batchDenom)
+	require.NoError(s.t, err)
+
+	balance, err := s.coreStore.BatchSupplyTable().Get(s.ctx, batch.Key)
+	require.NoError(s.t, err)
+
+	require.Equal(s.t, expected.RetiredAmount, balance.RetiredAmount)
+	require.Equal(s.t, expected.TradableAmount, balance.TradableAmount)
+}
+
+func (s *buyDirectSuite) sellOrderSetup(count int) {
```

Comment:
could we have a comment on what `count` is for? i was a bit confused at first

Reply:
https://github.com/regen-network/regen-ledger/pull/1122/commits/023fd694b90e82d70b1646d70f3e20dd5043d5fe